docs/rules/expiring-todo-comments.md

```diff
@@ -0,0 +1,286 @@
+- Expire when a **package** reaches a specific **version**.
+
+## Conditions
+
+### Expiry Date
+
+Using a date as condition, a TODO will only work as longs as this date
+is not met. This is specially useful when you either know when the
+action should take place or simply want to set boundaries for yourself.
+
+```js
+// TODO [2019-11-15]: Refactor this code before the sprint ends.
+// TODO (@lubien) [2019-07-18]: When John deliver his code. I can reuse it.
+// TODO [2019-08-10] I must refactor this for sure before I deliver.
+```
+
+Dates are [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601#Dates) and
+use [UTC](https://en.wikipedia.org/wiki/Coordinated_Universal_Time) timezone.
+
+Multiple dates on the same TODO are not valid and will be reported.
+There's no meaning in having two dates as the closest one will always be
+the one to report.
+
+### Package Version
+
+If you're working on a NPM package, you should have a root level package.json
+with a [`version`](https://docs.npmjs.com/files/package.json#version) field
+as specified from NPM itself. This condition takes place under this value.
+
+By assigning a condition to compare a package.json `version` the TODO will
+report as soon as the condition is met. It's really useful when the work
+is related to the version of the package itself such as deprecated APIs.
+
+```js
+// TODO [>=1.0.0]: I should work around this when we reach v1.
+// TODO @lubien [>0]: For now this is fine but for a stable version we must refactor.
+// FIXME [>10] This feature is deprecated and should be removed on next major version.
+```
+
+Multiple package versions on the same TODO are not valid and will be reported.
+Just like dates, multiple versions will make no sense as the closest one will
+be the one to trigger the report.
+
+### Engine Version
+
+On package.json you can specify some [engines](https://docs.npmjs.com/files/package.json#engines)
+that your code works on. With this setting you can specify both "node" and "npm" versions you can
+work on.
+
+With that in mind, this condition is triggered as soon as package.json engine reaches the target version.
+This is particularly useful for maintainers that strive for compatibility but have plans
+to the future where some feature will be widely available and the could drop support for
+older versions.
+
+Imagine you want to use async/await but for now you support node versions that don't have it?
+Maybe npm will have a feature that could be useful later but for now you have to keep support
+for older version users?
+
+```js
+// TODO [engine:node@>=8]: We can use async/await now.
+// FIXME [engine:node@>=20.0.0] Hey, node can use import/export now, we should refactor.
+// XXX @lubien [engine:npm@>4.0]: Now npm have [feature] we should use it.
+```
+
+Argument versions should be [semver](https://semver.org/) compatible such as: `1.2.45`, `5.3`, `1`.
+
+### Dependency Presence
+
+As a developer using NodeJS you should be know to package.json
```

Comment:
```suggestion
As a developer using Node.js, you should be know to package.json
```

Not sure what you're trying to say here.

Reply:
Rephrased

> As a Node.js developer, you probably already know about package.json dependencies and devDependencies.

Reply:
Done